aidants_connect_web/models.py

```diff
@@ -335,6 +335,23 @@ def has_a_carte_totp(self):
         except CarteTOTP.DoesNotExist:
             return False
 
+    def remove_from_organisation(self, organisation: Organisation) -> Optional[bool]:
+        if self.organisations.filter(pk=organisation.id).count() == 0:
+            return None
+
+        if self.organisations.count() == 1:
+            self.is_active = False
+            self.save()
+
+            return self.is_active
+
+        self.organisations.remove(organisation)
+        if self.organisations.filter(pk=self.organisation.id).count() == 0:
```

Comment:
J'ai dû faire ce test souvent, je le trouve plus lisible comme ceci : 

```suggestion
        if self.organisation not in self.organisations.all():
```

Reply:
Je pense que cette version est moins efficiente. Déjà, elle fait remonter toutes les relations en base alors que l'on ne cherche à certifier la présence que de l'une d'entre elles. Ensuite, c'est Python qui fait la recherche plutôt que la BDD, ce qui est plus gourmand en ressources.

Dans ce cas, c'est pas trop grave, parce qu'on s'attend à ce que le nombre d'organisations dont un aidant fait partie soit limité. Mais de manière générale, je commence à me dire que c'est pas forcément une bonne pratique d'utiliser `all()`.

Si c'est qu'une question de lisibilité, je peux écrire une fonction `Aidant.member_of(organisation: Organisation)` ,

Reply:
Je dirais que ce qu'on gagne en lisibilité est plus important que ce qu'on perd en efficacité machine. On n'aura pas 20k responsables et ils auront un usage modéré de ces pages-là (on n'ajoute pas des aidants 15 fois par jour).
Si tu préfères faire une fonction qui va chercher en BDD tout en restant lisible, on peut ! Mais juste attention à bien la tester y compris dans des cas un peu tricky comme celui-ci (j'aurais peur que l'on se fasse avoir par du cache BDD à la con).